pkg/api/login_oauth.go

```diff
@@ -210,3 +216,14 @@ func redirectWithError(ctx *middleware.Context, err error, v ...interface{}) {
 	ctx.Session.Set("loginError", err.Error())
 	ctx.Redirect(setting.AppSubUrl + "/login")
 }
+
+var logoutMap sync.Map
```

Comment:
Storing the logout url in memory won't work when running multiple Grafana instances since you might log in to one server and log out on another server. Not sure how to solve that yet. We are thinking about having redis as a requirement for multiple Grafana instances but that's not implemented yet.

Reply:
Is it fine to store in user cookie than ? See commit below.